x/subscription/README.md

```diff
@@ -0,0 +1,202 @@
+	PlanIndex      string // index (name) of plan
+	PlanBlock      uint64 // when the plan was created
+	DurationBought uint64 // total requested duration in months
+}
+```
+
+When a consumer buys a subscription, an admin project is automatically generated for them.  
+When the creator and consumer are different individuals, indicating that someone is purchasing a subscription for another user, the tokens will be taken out of the creator's account.
+
+Subscriptions are saved in a fixation store (see [FixationStore](https://github.com/lavanet/lava/blob/main/x/fixationstore/README.md) module for reference).
+
+There could only be a single active subscription for a consumer.  
+When a consumer purchase a subscription, the tokens are transferred from their account immediately.  
+Those tokens are later on distributed among the providers that served that consumer, and the delegators that staked those providers.
+
+### Advance Month
+
+One of the primary processes within this module involves the function `advanceMonth`, which is invoked upon the expiration of a timer set by the subscription. This function holds significance as it manages the logic that occurs at the conclusion of each month of a subscription.
+
+Here's a high-level flow of the function:
+
+1. **Update the CU Tracker Timer:**
+
+   - Add a CU tracker timer for the subscription.
+
+2. **Check Subscription Duration:**
+
+   - If the subscription’s remaining duration (`DurationLeft`) is zero:
+     - If this occurs, it's considered a bug. Log the issue and extend the subscription by one month for smoother operation and investigation.
+   - If there is still time left (`DurationLeft` > 0):
+     - reduce it by one to reflect the passage of a month.
+     - Increase the total duration (`DurationTotal`) by one.
+     - Reset and update the subscription details.
+   - If no time is left (`DurationLeft` = 0):
+     - Subscription is expired. Refer to the next step.
+
+3. **Handle Special Cases for Subscriptions:**
+
+   - If a future subscription is set:
+     - Activate the future subscription by updating current subscription details with those of the future subscription and restart the subscription.
+   - Else, if auto-renewal is enabled:
+     - Attempt to renew the subscription for another month.
+     - If renewal fails, remove the expired subscription.
+   - If there is no future subscription or auto-renewal, remove the expired subscription.
+
+### Subscription Upgrade
+
+A subscription can be upgraded to a more expensive plan.
+
+That also means, that if a consumer buys a plan, and then decides to downgrade their plan, the way to do it is either buy more months in the cheaper subscription, so the amount that the consumer will pay will be larger than the one using the same command as shown above, with the only caveat - the `plan-index` must be different than the currently active subscription's plan, and, it must be with a higher price.
```

Comment:
I really didn't understand this paragraph. can you rephrase it? also, i thought downgrading is not possible

Reply:
Downgrading is not possible, but there are ways to achieve it in an indirect way, which I explain in this section. 

How's this:
```
While direct downgrading of a subscription plan is not officially supported, consumers seeking a lower-tier plan can still achieve a similar effect. This can be done by purchasing additional months under a less expensive plan. It's important to note that the new plan's plan-index must differ from the currently active one, and the total payment for the extended period under the cheaper plan should exceed the amount of the original, higher-priced plan. This workaround allows consumers to effectively 'downgrade' their subscription within the system's constraints
```

Reply:
yeah, this is much clearer.
Just a question, this is kind of a "weird trick" to achieve something pretty common. Is this behavior wanted? why would we limit the auto-renewal mechanism like this? if downgrade is possible and not contradicting other parts of the code, why won't we let consumers buy cheaper plans with any duration?

Reply:
actually, looking at the code downgrading is not possible in any way. When you upgrade the plan, you check the plan price alone (without multiplying by the duration), so downgrading won't work

Reply:
Exactly. Currently it is not supported, simply because it's too complicated to implement, alongside the face that it is not really a must-have feature.

Rephrase the paragraph in 0209b1a3defe1341632864c7780f1adb17052c70